vespa-hadoop/src/main/java/com/yahoo/vespa/hadoop/pig/VespaDocumentOperation.java

```diff
@@ -309,9 +351,32 @@ private static boolean shouldWritePartialUpdate(Operation op, int depth) {
 
     private static void writePartialUpdate(Object value, Byte type, JsonGenerator g, String name, Properties properties, Schema schema, Operation op, int depth) throws IOException {
         g.writeStartObject();
-        g.writeFieldName(PARTIAL_UPDATE_ASSIGN); // TODO: lookup field name in a property to determine correct operation
+        // look up which operation to do by checking names and their respected properties
+        if (!isPartialOperation(REMOVE_TENSOR_FIELDS, name, properties, g, PARTIAL_UPDATE_REMOVE, true)
+        && !isPartialOperation(REMOVE_BAG_AS_MAP_FIELDS, name, properties, g, PARTIAL_UPDATE_REMOVE, true)
+                && !isPartialOperation(ADD_TENSOR_FIELDS, name, properties, g, PARTIAL_UPDATE_ADD, true)) {
+            g.writeFieldName(PARTIAL_UPDATE_ASSIGN);
+        }
         writeValue(value, type, g, name, properties, schema, op, depth);
         g.writeEndObject();
+
+
+    }
+
+    private static boolean isPartialOperation(String label, String name, Properties properties, JsonGenerator g, String targetOperation, boolean writeFieldName) throws IOException{
```

Comment:
Can we merge `isPartialOperation` into `writePartialUpdate`?
The main purpose of `isPartialOperation` is to write `assign` to the output.
And the purpose of `isPartialOperation` is to check it's a partial operation and write the operation to the output.
I think the purpose of these two method is alike.
It's better to write the same purpose code into same place.